### PR TITLE
Tin 2

### PR DESCRIPTION
**Pull request by Kr1s4an**
Jwt login

### CODE REVIEW

File: src/main/java/com/volasoftware/tinder/service/UserService.java [hunk 1]
```diff
@@ -107,4 +122,15 @@ public boolean isEmailRegistered(String email) {
...
+    public UserDetailsService userDetailsService() {
+        return new UserDetailsService() {
+            @Override
+            public UserDetails loadUserByUsername(String username) {
+                return userRepository.findOneByEmail(username).orElseThrow(() ->
+                        new UsernameNotFoundException("User not found"));
+            }
+        };
+    }
+
```
Comment (kzlatkovski) [inline src/main/java/com/volasoftware/tinder/service/UserService.java]: Remove unnecessary empty spaces.

File: src/main/java/com/volasoftware/tinder/controller/LoginController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        this.jwtService = jwtService;
+    }
+
+    @PostMapping("/api/v1/users/login")
+    public ResponseEntity loginUser(@Valid @RequestBody LoginUserDto loginUserDto) {
+        User authenticatedUser = userService.loginUser(loginUserDto);
+
+        String jwtToken = jwtService.generateToken(authenticatedUser);
+
+
```
Comment (kzlatkovski) [inline src/main/java/com/volasoftware/tinder/controller/LoginController.java]: Remove unnecessary empty spaces.

File: src/main/java/com/volasoftware/tinder/controller/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        this.userService = userService;
+        this.jwtService = jwtService;
+    }
+
+    @PostMapping("/api/v1/users/login")
+    public ResponseEntity loginUser(@Valid @RequestBody LoginUserDto loginUserDto) {
+        User authenticatedUser = userService.loginUser(loginUserDto);
+
+        String jwtToken = jwtService.generateToken(authenticatedUser);
+
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/controller/LoginController.java:29]: Remove extra line

File: src/main/java/com/volasoftware/tinder/jwt/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request,
+                                    @NonNull HttpServletResponse response,
+                                    @NonNull FilterChain filterChain)
+            throws ServletException, IOException {
+        final String authHeader = request.getHeader("Authorization");
+        final String jwt;
+        final String userEmail;
+        if (StringUtils.isEmpty(authHeader) || !StringUtils.startsWith(authHeader, "Bearer ")) {
```
Comment (kzlatkovski) [inline src/main/java/com/volasoftware/tinder/jwt/JwtAuthenticationFilter.java]: Extract Bearer into a constant.

File: src/main/java/com/volasoftware/tinder/jwt/JwtServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    public String generateToken(UserDetails userDetails){
+        return generateToken(new HashMap<>(), userDetails);
+    }
+
+    @Override
+    public boolean isTokenValid(String token, UserDetails userDetails) {
+        final String userName = extractUserName(token);
+        return (userName.equals(userDetails.getUsername())) && !isTokenExpired(token);
+    }
+    private <T> T extractClaim(String token, Function<Claims, T> claimsResolvers){
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/jwt/JwtServiceImpl.java:38]: For more readability, separate methods with new line

File: src/main/java/com/volasoftware/tinder/jwt/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            filterChain.doFilter(request, response);
+            return;
+        }
+        jwt = authHeader.substring(7);
```
Comment (kzlatkovski) [inline src/main/java/com/volasoftware/tinder/jwt/JwtAuthenticationFilter.java]: You can use string.lenght to get the size of the prefix that needs to be excluded

File: src/main/java/com/volasoftware/tinder/jwt/JwtServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        final Claims claims = extractAllClaims(token);
+        return claimsResolvers.apply(claims);
+    }
+    private String generateToken(Map<String, Object> extraClaims, UserDetails userDetails){
+        return Jwts.builder().setClaims(extraClaims).setSubject(userDetails.getUsername())
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * 24))
```
Comment (kzlatkovski) [inline src/main/java/com/volasoftware/tinder/jwt/JwtServiceImpl.java]: Extract the millisecond time representations into constants.

**Review by todornikolov1 — CHANGES_REQUESTED** [review]

**Pull request merged**